catalog/resource_volume.go

```diff
@@ -50,6 +53,25 @@ func ResourceVolume() common.Resource {
 				Type:     schema.TypeString,
 				Computed: true,
 			}
+			// As of 3rd December 2024, the Volumes create API returns an incorrect
+			// error message "CreateVolume Missing required field: volume_type"
+			// if you specify an invalid value for volume_type (i.e. not one of "MANAGED" or "EXTERNAL").
+			//
+			// If server side validation is added in the future, this validation function
+			// can be removed.
+			m["volume_type"].ValidateDiagFunc = func(i interface{}, p cty.Path) diag.Diagnostics {
```

Comment:
why do we need complex validation function instead of `.SetValidateFunc(validation.StringInSlice`? like here: https://github.com/databricks/terraform-provider-databricks/blob/main/catalog/resource_workspace_binding.go#L46

Reply:
I'm fine with either approach. The advantage is if a new volume type is added in the future (unlikely, but who knows), then this code will automatically be updated.

Let me know which approach you prefer though.

Reply:
We need to generalize it - I discussed it with @mgyucht before - we have other places where we need to check against enum

Reply:
Thanks, switched to the simpler validation function here. This can be removed once we have the general mechanism.